xbout/plotting/animate.py

```diff
@@ -1,21 +1,129 @@
+    account branch cuts (X-points).
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        A 2D (x,y) DataArray of data to plot
+    ax : Axes, optional
+        A matplotlib axes instance to plot to. If None, create a new
+        figure and axes, and plot to that
+    separatrix : bool, optional
+        Add dashed lines showing separatrices
+    targets : bool, optional
+        Draw solid lines at the target surfaces
+    add_limiter_hatching : bool, optional
+        Draw hatched areas at the targets
+    **kwargs : optional
+        Additional arguments are passed on to method
+
+    ###Returns
+    ###-------
+    ###artists
+    ###    List of the contourf instances
+    """
+
+    # TODO generalise this
+    x = kwargs.pop('x', 'R')
+    y = kwargs.pop('y', 'Z')
+
+    # Check plot is the right orientation
+    spatial_dims = list(da.dims)
+
+    try:
+        spatial_dims.remove(animate_over)
+    except ValueError:
+        raise ValueError("Dimension animate_over={} is not present in the data"
+                         .format(animate_over))
+
+    if len(da.dims) != 3:
+        raise ValueError("da must be 2+1D (t,x,y)")
+
+    if ax is None:
+        fig, ax = plt.subplots()
+    else:
+        fig = ax.get_figure()
+
+    if vmin is None:
+        vmin = da.min().values
+    if vmax is None:
+        vmax = da.max().values
+
+    # pass vmin and vmax through kwargs as they are not used for contour plots
+    kwargs['vmin'] = vmin
+    kwargs['vmax'] = vmax
+
+    # create colorbar
+    norm = (kwargs['norm'] if 'norm' in kwargs
+            else matplotlib.colors.Normalize(vmin=vmin, vmax=vmax))
+    sm = plt.cm.ScalarMappable(norm=norm, cmap=cmap)
+    sm.set_array([])
+    cmap = sm.get_cmap()
+    fig.colorbar(sm, ax=ax)
+
+    ax.set_aspect('equal')
+
+    regions = _decompose_regions(da)
+
+    # Plot all regions on same axis
+    blocks = []
+    for region in regions:
+        # Load values eagerly otherwise for some reason the plotting takes
+        # 100's of times longer - for some reason animatplot does not deal
+        # well with dask arrays!
+        blocks.append(amp.blocks.Pcolormesh(region.coords[x].values,
+                      region.coords[y].values, region.values, ax=ax, cmap=cmap,
+                      **kwargs))
+
+    ax.set_title(da.name)
+    ax.set_xlabel(x)
+    ax.set_ylabel(y)
+
+    if separatrix:
+        plot_separatrices(da, ax)
+
+    if targets:
+        plot_targets(da, ax, hatching=add_limiter_hatching)
+
+    if animate:
+        timeline = amp.Timeline(np.arange(da.sizes[animate_over]), fps=fps)
+        anim = amp.Animation(blocks, timeline)
+
+        if controls:
+            anim.controls(timeline_slider_args={'text': animate_over})
+
+        if not save_as:
+            save_as = "{}_over_{}".format(da.name, animate_over)
+        anim.save(save_as + '.gif', writer=PillowWriter(fps=fps))
+
+    return blocks
+
 from .utils import plot_separatrix
```

Comment:
`plot_separatrix` is no longer used?

Reply:
`plot_separatrix` isn't used in the library, but does a different thing from `plot_separatrices` - it plots a separatrix at a given x-position, and can be used with x-z plots, while `plot_separatrices` works on R-Z plots. It can usefully be called by the user to add a separatrix. It might be good to make the naming of the two functions less confusing, and possibly make `plot_separatrix` get the separatrix positions from `ixseps1` and `ixseps2` (if available).

Reply:
It used to be used by `animate_imshow`, I just meant it doesn't need to imported anymore. This was solved by https://github.com/boutproject/xBOUT/pull/38/commits/85b0cc6ebe816604b5994662e99c5c33ac35d6f0 .